.github/workflows/emacs-lint.yml

```
@@ -15,20 +15,40 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
+        os:
+          - ubuntu-latest
+        ocaml-compiler:
+          - 5.2.x
         emacs_version:
           - '27.2'
           - '28.2'
-          - '29.1'
+          - '29.3'
           - snapshot
       fail-fast: false  # don't stop jobs if one fails
     env:
       EMACS_PACKAGE_LINT_IGNORE: ${{ matrix.package_lint_ignore }}
       EMACS_BYTECOMP_WARN_IGNORE: ${{ matrix.bytecomp_warn_ignore }}
     steps:
-    - uses: purcell/setup-emacs@v6.0
-      with:
-        version: ${{ matrix.emacs_version }}
+      - uses: purcell/setup-emacs@v6.0
```

```suggestion
      - uses: purcell/setup-emacs@master
```
I usually recommend that people not pin this to a specific version, as the underlying Emacs (and Nix installer) versions change frequently.